src/devices/src/virtio/balloon/device.rs

```diff
@@ -441,6 +441,16 @@ impl Balloon {
             return Err(BalloonError::StatisticsStateChange);
         }
 
+        let mem = mem_of_active_device!(self.device_state);
+        if let Some(index) = self.stats_desc_index.take() {
+            self.queues[STATS_INDEX]
+                .add_used(&mem, index, 0)
+                .map_err(BalloonError::Queue)?;
+            self.signal_used_queue()?;
+        } else {
+            error!("Could not update balloon statistics.");
+        }
```

Comment:
Please move this to a specialized `fn trigger_stats(&mut self)` and call it both from here and `process_stats_timer_event()`